tests/cpp20_modules/main.cpp

```
@@ -0,0 +1,14 @@
+#include <iostream>
+
+import proxy;
+import foo;
+import foo_impl;
+
+auto user(pro::proxy<Foo> p) {
+  std::cout << "Foo says: " << p->GetFoo() << '\n';
+}
+
+int main() {
+  MyFoo foo;
+  user(&foo);
```

Not blocking this PR, but we should fail fast if `p->GetFoo()` is not `42`.

---

I think a better way here is to merge it into the main test target and the additional sources related to modules are conditionally enabled when building with modules.

---

Addressed in latest commit.